hail/src/main/scala/is/hail/expr/ir/MatrixWriter.scala

```diff
@@ -1126,28 +1127,24 @@ case class BGENPartitionWriter(typ: MatrixType, entriesFieldName: String, writeH
     entries.forEachDefinedOrMissing(cb)({ (cb, j) =>
       emitNullGP(cb)
     }, { case (cb, j, entry: SBaseStructValue) =>
-      if (entry.st.virtualType.fieldIdx.get("GP").isDefined) {
```

Comment:
Couldn't the GP field still be missing even though the entries aren't? (Meaning we need this if)

Reply:
will do the same for GEN though!